ocaml/xapi/context.ml

```diff
@@ -328,17 +323,9 @@ let start_tracing_helper ?(span_attributes = []) parent_fn task_name =
   in
   let span_name, span_attributes = span_details_from_task_name task_name in
   let parent = parent_fn span_name in
-  let span_kind = span_kind_of_parent parent in
-  let tracer = get_tracer ~name:span_name in
-  match
-    Tracer.start ~span_kind ~tracer ~attributes:span_attributes ~name:span_name
-      ~parent ()
-  with
-  | Ok x ->
-      x
-  | Error e ->
-      R.warn "Failed to start tracing: %s" (Printexc.to_string e) ;
-      None
+
+  Tracing.with_tracing ~attributes:span_attributes ~parent ~name:span_name
+    Fun.id
```

Comment:
Upon talking with @mg12 I realized the original behavior closes spans somewhere else. I'll have a fixup for this :)